app/pages/learn/01_tutorial/04_mastering-the-api/02_invoke/00_methodhandle.md

```diff
@@ -0,0 +1,531 @@
+MethodHandle replaceMethodHandle = lookup.findVirtual(String.class, "replace", replaceMethodType);
+String result = (String) replaceMethodHandle.invoke((Object)"dummy", (Object)'d', (Object)'m'); // would fail with `invokeExact`
+```
+
+One other alternative to invoke a method handle is to use `MethodHandle::invokeWithArguments`.
+The result of this method invocation is equivalent to `invoke`, with the only difference that all the arguments can be provided as an array or list of objects. 
+
+One interesting feature of this method is that if the number of provided arguments exceeds the expected number, 
+all the leftover arguments will be squashed into the last argument, which will be treated as an array.
+
+<a id="fields">&nbsp;</a>
+## Accessing fields
+It is possible to create method handles that have read or write access to fields. 
+For instance fields, this is facilitated by `findGetter` and `findSetter` methods, and for static fields, by `findStaticGetter` and `findStaticSetter` methods. 
+We don't need to provide a `MethodType` instance; instead, we should provide a single type, which is the type of the field.
+
+For example, if we have a static field `magic` in our `Example` class:
+
+```java
+private static String magic = "initial value static field";
+```
+
+Given that we have created a `Lookup` object
+
+```java
+MethodHandles.Lookup lookup = MethodHandles.lookup();
+```
+
+We can simply create both setter and getter method handles and invoke them separately:
+
+```java
+MethodHandle setterStaticMethodHandle = lookup.findStaticSetter(Example.class, "magic", String.class);
+MethodHandle getterStaticMethodHandle = lookup.findStaticGetter(Example.class, "magic", String.class);
+
+setterStaticMethodHandle.invoke("new value static field");
+String staticFieldResult = (String) getterStaticMethodHandle.invoke(); // staticFieldResult == `new value static field`
+```
+
+Here is an instance field `abc` of class `Example`:
+
+```java
+private String abc = "initial value";
+```
+
+We can similarly create method handles for reading and writing to the instance field:
+
+```java
+MethodHandle setterMethodHandle = lookup.findSetter(Example.class, "abc", String.class);
+MethodHandle getterMethodHandle = lookup.findGetter(Example.class, "abc", String.class);
+```
+
+To use setter and getter method handles with an instance field, we must first obtain an instance of the class where the field belongs:
+
+```java
+Example example = new Example();
+```
+
+Afterward, we must provide an instance of `Example` for invocation of our setter and getter:
+
+```java
+setterMethodHandle.invoke(example, "new value");
+String result = (String) getterMethodHandle.invoke(example); // result == `new value`
+```
+
+Although it is possible to read and write field values using method handles, it's not common practice. 
+For fields, it's more suitable to use `VarHandle`s instead, which can be created using `findVarHandle` and `findStaticVarHandle` methods.
```

Comment:
Since it's more common to use `VarHandle`s, why did you provide an example on accessing fields with `MethodHandle` but not with `VarHandle`?

Reply:
Wanted to show what's possible to do with `MethodHandle` in principle.
There will be a separate tutorial on `VarHandle` so I intentionally didn't write about it here.

Reply:
In that case, it might make sense to link to the `VarHandle` tutorial here whenever that gets added.